flocx_market/db/sqlalchemy/bid_api.py

```diff
@@ -0,0 +1,39 @@
+from flocx_market.db.orm import orm
+from flocx_market.db.sqlalchemy.bid_model import BidModel
+from uuid import uuid4
+import json
+
+
+class BidApi(BidModel):
+    def __init__(self, creator_bid_id, creator_id, server_quantity, start_time, end_time, duration, status, server_config_query, cost):
+        self.marketplace_bid_id = str(uuid4())
+        self.creator_bid_id = creator_bid_id
+        self.creator_id = creator_id
+        self.server_quantity = server_quantity
+        self.start_time = start_time
+        self.end_time = end_time
+        self.duration = duration
+        self.status = status
+        self.server_config_query = server_config_query
+        self.cost = cost
```

Comment:
Let's drop this `__init__` method here just like in #e49134845c4d754f01b1c5d5fdd860cf27fccca4. The one provided by the underlying `*Model` object is more useful.

Reply:
the only thing with removing this is that then the marketplace_offer_id is expected as an argument. If we keep it in there, we generate the marketplace_offer_id as a UUID during the init method. How would you suggest resolving that if we remove it? @larsks

Reply:
Something like this: https://github.com/CCI-MOC/esi-leap/blob/master/esi_leap/db/sqlalchemy/api.py#L166

Reply:
I'm not a 100% sure if removing one method to add another one is the move? and I don't know how to do the UUID generation without it

Reply:
Having a specific `create` method that also lets you set defaults makes a lot of sense to me. It seems odd to have the `__init__` method of BidApi be the equivalent of creating a new Bid.

UUID generation can be handled through oslo utils, as seen in the linked code. In general, whenever there's an oslo library that we can use, we should *always* be using it.

Reply:
oh okay that makes sense @tzumainn. I just realized that Lars used a different approach here: https://github.com/CCI-MOC/flocx-market/commit/e49134845c4d754f01b1c5d5fdd860cf27fccca4

what do you think?

Reply:
@larsks sorry for the confusion there, I completely missed that you were generating the UUID on the model itself. It makes way more sense that way

Reply:
@fvukelic I'd personally prefer for it to get in the api, and not model. We're also going to end up doing things such as defaulting project_id (taken from the context) and the like. But I don't have a strong opinion on whether that has to be done right now.

Reply:
@tzumainn I kind of want default values in the model. Let me know if you have any free time tomorrow morning and we can chat?

Reply:
@larsks Sure! I've seen some interesting things in the OpenStack code, and would not mind your opinion on them.